diff -rupN 2009_0302_RT2870_Linux_STA_v2.1.0.0.old/os/linux/config.mk 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/config.mk
--- 2009_0302_RT2870_Linux_STA_v2.1.0.0.old/os/linux/config.mk	2009-03-10 00:47:50.000000000 -0400
+++ 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/config.mk	2009-03-26 01:57:12.000000000 -0400
@@ -236,14 +236,14 @@ export CFLAGS
 endif
 
 ifeq ($(PLATFORM),PC)
-    ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+	#    ifneq (,$(findstring 2.4,$(LINUX_SRC)))
 	# Linux 2.4
-	CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -O2 -fomit-frame-pointer -fno-strict-aliasing -fno-common -pipe -mpreferred-stack-boundary=2 -march=i686 -DMODULE -DMODVERSIONS -include $(LINUX_SRC)/include/linux/modversions.h $(WFLAGS)
-	export CFLAGS
-    else
+	#	CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -O2 -fomit-frame-pointer -fno-strict-aliasing -fno-common -pipe -mpreferred-stack-boundary=2 -march=i686 -DMODULE -DMODVERSIONS -include $(LINUX_SRC)/include/linux/modversions.h $(WFLAGS)
+	#	export CFLAGS
+	#    else
 	# Linux 2.6
 	EXTRA_CFLAGS := $(WFLAGS) -I$(RT28xx_DIR)/include
-    endif
+	#    endif
 endif
 
 ifeq ($(PLATFORM),IXP)
