diff -rupN old/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/2870_main_dev.c new/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/2870_main_dev.c
--- old/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/2870_main_dev.c	2008-12-24 02:34:46.000000000 -0500
+++ new/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/2870_main_dev.c	2009-01-15 12:30:46.000000000 -0500
@@ -962,7 +962,7 @@ VOID RT28xxThreadTerminate(
 		pAd->mlme_kill = 1;
 		//RT28XX_MLME_HANDLER(pAd);
 		mb();
-		ret = kill_proc (pObj->MLMEThr_pid, SIGTERM, 1);
+		ret = kill_pid (pObj->MLMEThr_pid, SIGTERM, 1);
 		if (ret) 
 		{
 			printk (KERN_WARNING "%s: unable to Mlme thread, pid=%d, ret=%d!\n",
@@ -985,7 +985,7 @@ VOID RT28xxThreadTerminate(
 		NdisReleaseSpinLock(&pAd->CmdQLock);
 		mb();
 		//RTUSBCMDUp(pAd);
-		ret = kill_proc(pObj->RTUSBCmdThr_pid, SIGTERM, 1);
+		ret = kill_pid(pObj->RTUSBCmdThr_pid, SIGTERM, 1);
 		if (ret) 
 		{
 			printk(KERN_WARNING "%s: unable to RTUSBCmd thread, pid=%d, ret=%d!\n",
@@ -1006,7 +1006,7 @@ VOID RT28xxThreadTerminate(
 		mb();
 		pAd->TimerFunc_kill = 1;
 		mb();
-		ret = kill_proc(pObj->TimerQThr_pid, SIGTERM, 1);
+		ret = kill_pid(pObj->TimerQThr_pid, SIGTERM, 1);
 		if (ret)
 		{
 			printk(KERN_WARNING "%s: unable to stop TimerQThread, pid=%d, ret=%d!\n", 
diff -rupN old/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/sta_ioctl.c new/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/sta_ioctl.c
--- old/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/sta_ioctl.c	2009-01-15 12:16:52.000000000 -0500
+++ new/2008_1225_RT3070_Linux_STA_v2.0.1.0/os/linux/sta_ioctl.c	2009-01-15 12:13:42.000000000 -0500
@@ -1113,7 +1113,7 @@ int rt_ioctl_giwscan(struct net_device *
 				memcpy(iwe.u.ap_addr.sa_data, &pAdapter->ScanTab.BssEntry[i].Bssid, ETH_ALEN);
 
         previous_ev = current_ev;
-			current_ev = iwe_stream_add_event(current_ev,end_buf, &iwe, IW_EV_ADDR_LEN);
+	current_ev = iwe_stream_add_event(info,current_ev,end_buf, &iwe, IW_EV_ADDR_LEN);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1192,7 +1192,7 @@ int rt_ioctl_giwscan(struct net_device *
 	}
 
 		previous_ev = current_ev;
-		current_ev	 = iwe_stream_add_event(current_ev,end_buf, &iwe, IW_EV_ADDR_LEN);
+		current_ev	 = iwe_stream_add_event(info,current_ev,end_buf, &iwe, IW_EV_ADDR_LEN);
 		if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
 	   		return -E2BIG;
@@ -1208,7 +1208,7 @@ int rt_ioctl_giwscan(struct net_device *
 		iwe.u.data.flags = 1;
  
         previous_ev = current_ev;
-		current_ev = iwe_stream_add_point(current_ev,end_buf, &iwe, pAdapter->ScanTab.BssEntry[i].Ssid);
+	current_ev = iwe_stream_add_point(info,current_ev,end_buf, &iwe, pAdapter->ScanTab.BssEntry[i].Ssid);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1235,7 +1235,7 @@ int rt_ioctl_giwscan(struct net_device *
 		iwe.len = IW_EV_UINT_LEN;
 
         previous_ev = current_ev;
-		current_ev = iwe_stream_add_event(current_ev, end_buf, &iwe,  IW_EV_UINT_LEN);
+	current_ev = iwe_stream_add_event(info,current_ev, end_buf, &iwe,  IW_EV_UINT_LEN);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1255,7 +1255,7 @@ int rt_ioctl_giwscan(struct net_device *
 		iwe.u.freq.i = 0;
 		
 		previous_ev = current_ev;
-		current_ev = iwe_stream_add_event(current_ev,end_buf, &iwe, IW_EV_FREQ_LEN);
+		current_ev = iwe_stream_add_event(info,current_ev,end_buf, &iwe, IW_EV_FREQ_LEN);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1270,7 +1270,7 @@ int rt_ioctl_giwscan(struct net_device *
     	iwe.u.qual.level = 0;
     	iwe.u.qual.noise = 0;
         set_quality(pAdapter, &iwe.u.qual, pAdapter->ScanTab.BssEntry[i].Rssi);
-    	current_ev = iwe_stream_add_event(current_ev, end_buf, &iwe, IW_EV_QUAL_LEN);
+    	current_ev = iwe_stream_add_event(info,current_ev, end_buf, &iwe, IW_EV_QUAL_LEN);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1288,7 +1288,7 @@ int rt_ioctl_giwscan(struct net_device *
 			iwe.u.data.flags = IW_ENCODE_DISABLED;
 
         previous_ev = current_ev;		
-        current_ev = iwe_stream_add_point(current_ev, end_buf,&iwe, (char *)pAdapter->SharedKey[BSS0][(iwe.u.data.flags & IW_ENCODE_INDEX)-1].Key);
+        current_ev = iwe_stream_add_point(info,current_ev, end_buf,&iwe, (char *)pAdapter->SharedKey[BSS0][(iwe.u.data.flags & IW_ENCODE_INDEX)-1].Key);
         if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
             return -E2BIG;
@@ -1316,7 +1316,7 @@ int rt_ioctl_giwscan(struct net_device *
     		    iwe.u.bitrate.value =  (tmpRate/2) * 1000000;
             
 			iwe.u.bitrate.disabled = 0;
-			current_val = iwe_stream_add_value(current_ev,
+			current_val = iwe_stream_add_value(info,current_ev,
 				current_val, end_buf, &iwe,
     			IW_EV_PARAM_LEN);            
 
@@ -1340,7 +1340,7 @@ int rt_ioctl_giwscan(struct net_device *
 						   pAdapter->ScanTab.BssEntry[i].WpaIE.IELen);
 			iwe.cmd = IWEVGENIE;
 			iwe.u.data.length = pAdapter->ScanTab.BssEntry[i].WpaIE.IELen;
-			current_ev = iwe_stream_add_point(current_ev, end_buf, &iwe, custom);
+			current_ev = iwe_stream_add_point(info,current_ev, end_buf, &iwe, custom);
 			if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
                 return -E2BIG;
@@ -1358,7 +1358,7 @@ int rt_ioctl_giwscan(struct net_device *
 						   pAdapter->ScanTab.BssEntry[i].RsnIE.IELen);
 			iwe.cmd = IWEVGENIE;
 			iwe.u.data.length = pAdapter->ScanTab.BssEntry[i].RsnIE.IELen;
-			current_ev = iwe_stream_add_point(current_ev, end_buf, &iwe, custom);
+			current_ev = iwe_stream_add_point(info,current_ev, end_buf, &iwe, custom);
 			if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
                 return -E2BIG;
@@ -1379,7 +1379,7 @@ int rt_ioctl_giwscan(struct net_device *
             for (idx = 0; idx < pAdapter->ScanTab.BssEntry[i].WpaIE.IELen; idx++)
                 sprintf(custom, "%s%02x", custom, pAdapter->ScanTab.BssEntry[i].WpaIE.IE[idx]);
             previous_ev = current_ev;
-    		current_ev = iwe_stream_add_point(current_ev, end_buf, &iwe,  custom);
+	    current_ev = iwe_stream_add_point(info,current_ev, end_buf, &iwe,  custom);
             if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
                 return -E2BIG;
@@ -1399,7 +1399,7 @@ int rt_ioctl_giwscan(struct net_device *
 			for (idx = 0; idx < pAdapter->ScanTab.BssEntry[i].RsnIE.IELen; idx++)
                 sprintf(custom, "%s%02x", custom, pAdapter->ScanTab.BssEntry[i].RsnIE.IE[idx]);
             previous_ev = current_ev;
-    		current_ev = iwe_stream_add_point(current_ev, end_buf, &iwe,  custom);
+	    current_ev = iwe_stream_add_point(info,current_ev, end_buf, &iwe,  custom);
             if (current_ev == previous_ev)
 #if WIRELESS_EXT >= 17
                 return -E2BIG;
